app/code/Bulbulatory/Recommendations/Controller/Recommendations/Add.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+
+namespace Bulbulatory\Recommendations\Controller\Recommendations;
+
+
+use Bulbulatory\Recommendations\Helper\RecommendationsHelper;
+use Bulbulatory\Recommendations\Model\RecommendationRepository;
+use Exception;
+use Magento\Customer\Model\Customer;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpPostActionInterface as HttpPostActionInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\UrlInterface;
+use Ramsey\Uuid\Uuid;
+
+/**
+ * Class Add
+ * @package Bulbulatory\Recommendations\Controller\Recommendations
+ */
+class Add extends LoggedInAction implements HttpPostActionInterface
+{
+    const ROUTE = 'customer/recommendations/add';
+    const POST_PARAM_EMAIL = 'recommendedEmail';
+    const XML_PATH_EMAIL_TEMPLATE_FIELD = 'recommendations/general/recommendation_email';
+
+    /**
+     * @var RecommendationRepository
+     */
+    private $recommendationsRepository;
+    /**
+     * @var UrlInterface
+     */
+    private $urlBuilder;
+
+    public function __construct(
+        Context $context,
+        RecommendationsHelper $recommendationsHelper,
+        Session $customerSession,
+        RecommendationRepository $recommendationRepository,
+        UrlInterface $urlBuilder
+    )
+    {
+        $this->urlBuilder = $urlBuilder;
+        $this->recommendationsRepository = $recommendationRepository;
+        parent::__construct($context, $recommendationsHelper, $customerSession);
+    }
+
+    /**
+     * @return ResultInterface
+     */
+    protected function _execute(): ResultInterface
+    {
+        if ($this->isRecommendationModuleEnabled()) {
+            $customer = $this->customerSession->getCustomer();
+            $recommendedEmail = $this->getRequest()->getParam(self::POST_PARAM_EMAIL);
+
+            if (empty($recommendedEmail)) {
+                $this->messageManager->addErrorMessage(__('Email field cannot be empty!'));
+            } elseif ($recommendedEmail === $customer->getEmail()) {
+                $this->messageManager->addErrorMessage(__('Recommended email cannot be same as yours!'));
+            } elseif ($this->recommendationsRepository->isRecommendationExist($customer, $recommendedEmail)) {
+                $this->messageManager->addErrorMessage(__('You already created recommendation for given email address!'));
+            } elseif (!$this->createRecommendation($customer, $recommendedEmail)) {
+                $this->messageManager->addErrorMessage(__('An error occurred while sending recommendation. please try again.'));
+            } else {
+                $this->messageManager->addSuccessMessage(__('Recommendation send successfully!'));
+            }
+        }
+        return $this->resultRedirectFactory->create()->setPath(Index::ROUTE);
```

Comment:
W `Index` jeśli moduł jest wyłączony, to nastąpi przekierowanie na stronę główną - można by je od razu tutaj ustawić, żeby tylko raz przekierować użytkownika, a nie ładować kilka akcji żeby docelowo trafił i tak w to samo miejsce.